docs/miscellaneous.rst

```diff
@@ -88,6 +88,234 @@ by checking if the lowest bit is set: short (not set) and long (set).
 .. note::
   Handling invalidly encoded slots is currently not supported but may be added in the future.
 
+JSON Output
+===========
+
+The storage layout of a contract can be requested via the :ref:`standard JSON interface <compiler-api>`.  The output is a JSON object containing two keys,
+``storage`` and ``types``.  The ``storage`` object is an array where each
+element has the following form:
+
+
+.. code::
+
+
+    {
+        "ast_id":"2",
+        "contract":"A",
+        "label":"x",
+        "offset":"0",
+        "slot":"0",
+        "type":"t_uint256"
+    }
+
+where the example above is the storage layout of ``contract A { uint x; }``
+and
+
+- ``ast_id`` is the id of the AST node of the state variable
+- ``contract`` is the name of the contract
+- ``label`` is the name of the state variable
+- ``offset`` is the offset within the storage slot according to the encoding
+- ``slot`` is the storage slot where the state variable resides
+- ``type`` is an identifier used as key to the variable's type information
+
+The given ``type``, in this case ``t_uint256`` represents an element in
+``types``, which has the form:
+
+
+.. code::
+
+    {
+        "encoding":"inplace",
+        "label":"uint256",
+        "numberOfBytes":"32",
+        "numberOfSlots":"1"
+    }
+
+where
+
+- ``encoding`` tells how the data is stored
+- ``label`` is the canonical type name
+- ``numberOfBytes`` the number of bytes that this type occupies within a slot
+- ``numberOfSlots`` the number of slots that this type requires in storage
+
+Some types have extra information besides the four above. Mappings contain
```

Comment:
You should probably list all of this, so including arrays with ``base`` and relate that to the information for ``encoding`` above and specify every additional field separately... I mean it's not like "some types have something extra", but that there are two kinds of fields: general fields and fields specific to the concrete encoding given by the value of ``encoding``, right?

I mean we could postpone all this for later, but it should actually be fully documented and not with an implicit "just try and figure it out" ;-).

Reply:
These are all the extra ones.

Reply:
I'm not sure I understand how this relates to the `encoding` field. Mapping and Arrays have extra keys and different encoding, but Structs have extra keys but the encoding is the same as value types.

Reply:
I somehow missed the fact that ``base`` is in fact in the next line, sorry... and true about encoding not actually relating - still: I wonder, if there *should* be something relating to it, like a ``kind`` field being ``value``, ``tuple``/``struct``, ``array`` or ``mapping``... having to look for the presence of any of the extra keys to tell what it is, is not nice, is it?

Reply:
But yeah - maybe I should read through the whole discussion about this before making random claims :-D.

Reply:
I was thinking about e.g. what happens if I forgot that the ``members`` key might exist (or more realistically we add another sub-type in the future)... then I'd just consider the thing as a regular value type... but actually this is just fine... and may be for all imaginable such cases... Yeah, ok, fine ;-).